presto-raptor/src/main/java/com/facebook/presto/raptor/util/DatabaseUtil.java

```diff
@@ -134,16 +140,26 @@ public static boolean isSyntaxOrAccessError(Exception e)
         return sqlCodeStartsWith(e, "42");
     }
 
+    public static boolean isTransactionCacheFullError(Exception e)
+    {
+        return sqlErrorCodeMatches(e, ER_TRANS_CACHE_FULL);
+    }
+
+    private static boolean sqlErrorCodeMatches(Exception e, int errorCode)
+    {
+        return Throwables.getCausalChain(e).stream().anyMatch(t -> (t instanceof SQLException) && ((SQLException) t).getErrorCode() == errorCode);
```

Comment:
I just realized that only matching on the error code isn't a good approach, since there can be different JDBC drivers which might happen to use the same error code number but for different purposes.